### PR TITLE
Add a Charge timer

### PR DESCRIPTION
**Pull request by Hydrox6**
Fulfils #3526 

Gif of it in action ([High quality version here](https://gfycat.com/JadedDecisiveBoubou))

### CODE REVIEW

**Comment by Infinitay** [comment]
If we don't have it already, can you add magic imbue?

**Comment by Hydrox6** [comment]
@Infinitay that's a thing already

File: runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersPlugin.java [hunk 1]
```diff
@@ -121,6 +122,11 @@ public void onVarbitChange(VarbitChanged event)
 	@Subscribe
 	public void updateConfig(ConfigChanged event)
 	{
+		if (!config.showCharge())
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersPlugin.java]: I would keep this ordered in the same way the config is ordered

Comment (Hydrox6) [reply runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersPlugin.java]: Decided to reorder the entire function, as it wasn't in order anyway

**Comment by josharoo** [comment]
For some reason my charge timer is resetting to 5:59 every time I use the home teleport

**Comment by Nightfirecat** [comment]
Is it because Charge uses the same graphic as normal spellbook teleports? If that's the case, it might be more reliable to check for the server message that's sent when you cast the spell: `You feel charged with magic power.`

**Comment by Hydrox6** [comment]
@Nightfirecat already implemented a fix in #3593

**Pull request merged**